tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -61,6 +61,53 @@ bool TypeTranslator::isRelaxedPrecisionType(QualType type) {
   return false;
 }
 
+bool TypeTranslator::isOpaqueType(QualType type) {
+  if (const auto *recordType = type->getAs<RecordType>()) {
+    const auto name = recordType->getDecl()->getName();
+
+    if (name == "Texture1D" || name == "RWTexture1D")
```

Comment:
Why are you not combining all these cases into one conditional? (or use a switch?)

Reply:
Just copied what the [existing code](https://github.com/Microsoft/DirectXShaderCompiler/blob/11dd70c4105012313f6e2d0c1a949ad4e94078d4/tools/clang/lib/AST/HlslTypes.cpp#L342) base does. I think it's more easier to read this way. Actually a `StringSwitch` seems nice for this. (We cannot use C `switch` on strings.)